Sources/RInAppMessaging/Router.swift

```diff
@@ -124,11 +142,90 @@ internal class Router: RouterType {
         return nil
     }
 
+    func displayTooltip(_ tooltipData: TooltipData,
+                        targetView: UIView,
+                        identifier: String,
+                        imageBlob: Data,
+                        becameVisibleHandler: @escaping (_ tooltipView: TooltipView) -> Void,
+                        completion: @escaping () -> Void) {
+
+        DispatchQueue.main.async {
+            self.displayedTooltips[identifier]?.removeFromSuperview()
+            self.displayedTooltips[identifier] = nil
+
+            let position = tooltipData.bodyData.position
+            let tooltipView = TooltipView(
+                model: TooltipViewModel(
+                position: position,
+                image: UIImage(data: imageBlob)!,
+                backgroundColor: UIColor(hexString: tooltipData.backgroundColor) ?? .white))
+
+            let onClose = {
+                self.displayedTooltips[identifier]?.removeFromSuperview()
+                self.displayedTooltips[identifier] = nil
+                completion()
+            }
+
+            tooltipView.onImageTap = {
+                if let uriToOpen = URL(string: tooltipData.bodyData.redirectURL ?? "") {
+                    UIApplication.shared.open(uriToOpen)
+                    onClose()
+                }
+            }
+            tooltipView.onExitButtonTap = onClose
+
+            let superview = self.findParentViewForTooltip(targetView)
+            guard superview != targetView else {
+                Logger.debug("Cannot find suitable view for tooltip targeting: \(targetView.description)")
```

Comment:
Maybe this should go to the error delegate?

Reply:
good idea